modules/social_features/social_group/modules/social_group_request/src/Plugin/ActivityAction/ChangeStatusGroupMembershipRequestActivityAction.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Drupal\social_group_request\Plugin\ActivityAction;
+
+use Drupal\activity_creator\Plugin\ActivityActionBase;
+
+/**
+ * Provides 'ChangeStatusGroupMembershipRequestActivityAction' activity action.
+ *
+ * @ActivityAction(
+ *  id = "change_status_group_membership_request_entity_action",
+ *  label = @Translation("Action that is triggered when a status of group membership request is changed"),
+ * )
+ */
+class ChangeStatusGroupMembershipRequestActivityAction extends ActivityActionBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isValidEntity($entity) {
+    /** @var \Drupal\Core\Entity\EntityInterface $entity */
+    return $entity->getEntityTypeId() === 'group_content';
```

Comment:
I was looking at your code to get inspiration for the event request enrollment feature and wondered why you use a custom action for this? For example in the actions there are also the onces called "Group content: Group membership request" and "Group content: Group membership", or didn't these one suit your needs?

Reply:
Maybe good to check Robert's question. If we can re-use something existing without the need for an additional action, that's probably better.

Reply:
@robertragas, @bramtenhove, I created new activity action because I needed to have message template when group manager approve membership request. So...yes you are right. I can use "Group content: Group membership" and implement needed ActivityContext. Will rework this

Reply:
Reworked. Check, please